File: frontend/src/lib/devReset.ts
// Dev/test-only reset API calls (thin wrappers over apiFetch).
// System-admin only; the backend also refuses these in production and requires an
// exact typed confirmation phrase.

import { apiFetch } from '@/lib/api'

export interface ResetCounts {
  imports: Record<string, number>
  live_crm: Record<string, number>
}

export interface ResetResult {
  action: string
  deleted: Record<string, number>
}

export function getResetCounts(): Promise<ResetCounts> {
  return apiFetch<ResetCounts>('/dev/reset/counts')
}

export function clearImports(confirm: string): Promise<ResetResult> {
  return apiFetch<ResetResult>('/dev/reset/imports', { method: 'POST', body: { confirm } })
}

export function clearLiveCrm(confirm: string): Promise<ResetResult> {
  return apiFetch<ResetResult>('/dev/reset/live-crm', { method: 'POST', body: { confirm } })
}

File: backend/app/schemas/dev_reset.py
"""Schemas for the dev/test-only reset tools."""
from __future__ import annotations

from pydantic import BaseModel, Field


class ResetCountsRead(BaseModel):
    """Preview of what each reset action would affect (read-only)."""

    imports: dict[str, int]
    live_crm: dict[str, int]


class ResetConfirm(BaseModel):
    """Body for a reset action — the exact typed confirmation phrase."""

    confirm: str = Field(..., min_length=1, max_length=64)


class ResetResult(BaseModel):
    """Result of a reset action: which action ran + per-table affected counts."""

    action: str
    deleted: dict[str, int]
